### DISCUSSION TITLE
.NET MAUI Windows desktop app with Azure AD B2C Authentication

### DISCUSSION THREAD

**Discussion by ajitanil** [dotnet/maui#10725] [Q&A]
I am developing a .NET MAUI WINDOWS desktop application.
Need to integrate the Azure AD B2C authentication for my .NET MAUI app.
What I found on the internet was bits and pieces of code which is not helpful.

Can anyone please share me a simple **sample code of .NET MAUI with Azure AD B2C for Windows App**?
Thank you in advance.

**Comment by davidbritch** [comment]
https://devblogs.microsoft.com/dotnet/authentication-in-dotnet-maui-apps-msal/

**Reply by ajitanil** [reply]
Thank you @davidbritch for your response.
The blog is very helpful.

However I am facing "API Permission" issues in my client application.
The API which is exposed is throwing some permission issues.

> AADSTS650052: The app is trying to access a service '872e6aed-61f2-4697-a4a0-ccede2d745ff'(AzureB2CAPI) that your organization 'e4658900-a1e3-4605-9c20-f3b25d4ecb74' lacks a service principal for. Contact your IT Admin to review the configuration of your service subscriptions or consent to the application in order to create the required service principal. 

Can you please let me know which permissions I need to add in  my client application?
Thank you :)
